src/obi_auth/util.py

```diff
@@ -0,0 +1,42 @@
+"""Utilities."""
+
+import base64
+import hashlib
+import platform
+import socket
+from pathlib import Path
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+
+
+def get_machine_salt():
+    """Get machine specific salt."""
+    hostname = socket.gethostname()
+    system = platform.uname().system
+    node = platform.uname().node
+    raw = f"{hostname}-{system}-{node}"
+    return hashlib.sha256(raw.encode()).digest()
+
+
+def derive_fernet_key() -> bytes:
+    """Create Fernet key from unique machine salt."""
+    hkdf = HKDF(
+        algorithm=hashes.SHA256(),
+        length=32,
+        backend=default_backend(),
+        salt=None,  # Optional: use one if you want context separation
+        info=b"machine-specific-fernet-key",  # Application-specific context
+    )
+    key = hkdf.derive(get_machine_salt())
+    return base64.urlsafe_b64encode(key)  # Fernet requires base64 encoding
+
+
+def get_config_path() -> Path:
+    """Get config file path."""
+    file_name = f"{get_machine_salt().hex()}.json"
```

Comment:
I'm wondering why the filename is named as the result of get_machine_salt().
If the salt is unique per machine, wouldn't be the same if it's a fixed name like `token.json`?
It would be even better if it can include the environment name, like `token-staging.json` and `token-production.json`

Reply:
Good point. Made the filename fixed based on the environment selected.